### PR TITLE
Java mongo upgrade version

### PR DESCRIPTION
**Pull request by flashboss**
Hi, I have an upgrade to use the last versions of mongodb

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -60,48 +61,47 @@ signing {
...
+					license {
+						name = 'The Apache License, Version 2.0'
+						url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+					}
+				}
+				developers {
+					developer {
+						id = 'roberttaylor426'
+						name = 'Robert Taylor'
+						email = 'john.doe@example.com'
```
Comment (roberttaylor426) [inline build.gradle:83]: You can omit my email address :-)

File: build.gradle [hunk 2]
```diff
@@ -60,48 +61,47 @@ signing {
...
+					connection = 'scm:git@github.com:sourcemuse/GradleMongoPlugin.git'
+					developerConnection = 'scm:git@github.com:sourcemuse/GradleMongoPlugin.git'
+					url = 'scm:git@github.com:sourcemuse/GradleMongoPlugin.git'
+				}
+			}
+		}
+    }	
     repositories {
-        mavenDeployer {
-            beforeDeployment { signing.signPom(it) }
```
Comment (roberttaylor426) [inline build.gradle:66]: Have we lost anything by no longer signing the POM in this way?

(Happy to figure this out myself when it comes to publication if you prefer)

File: src/main/groovy/com/sourcemuse/gradle/plugin/flapdoodle/adapters/CustomFlapdoodleRuntimeConfig.groovy [hunk 3]
```diff
@@ -51,16 +54,22 @@ class CustomFlapdoodleRuntimeConfig extends RuntimeConfigBuilder {
...
             }
         })
+		
+		Builder builder = ArtifactStore.builder();
+		builder.downloadConfig(downloadConfigBuilder.build())
+		builder.downloader(Downloader.platformDefault())
+		builder.tempDirFactory(downloadConfigBuilder.artifactStorePath)
+		builder.executableNaming(downloadConfigBuilder.fileNaming)
+		ImmutableArtifactStore artifactStore = builder.build()
+        runtimeConfigBuilder.artifactStore(artifactStore)
```
Comment (roberttaylor426) [inline src/main/groovy/com/sourcemuse/gradle/plugin/flapdoodle/adapters/CustomFlapdoodleRuntimeConfig.groovy:71]: _Super_-nitpicky - but could we keep indentation consistent throughout this PR?

File: src/test/groovy/com/sourcemuse/gradle/plugin/BuildScriptBuilder.groovy [hunk 4]
```diff
@@ -1,8 +1,8 @@
-package com.sourcemuse.gradle.plugin
+ package com.sourcemuse.gradle.plugin
 
 class BuildScriptBuilder {
 
-    static final DEFAULT_MONGOD_PORT = 27017
+    static final public DEFAULT_MONGOD_PORT = 27017
```
Comment (roberttaylor426) [inline src/test/groovy/com/sourcemuse/gradle/plugin/BuildScriptBuilder.groovy:5]: What was the reason for this? Just curious

File: src/test/groovy/com/sourcemuse/gradle/plugin/MongoPluginConfigSpec.groovy [hunk 5]
```diff
@@ -1,25 +1,27 @@
...
 
 class MongoPluginConfigSpec extends Specification {
 
-    def static final VERBOSE_LOGGING_SAMPLE = 'isMaster'
+    def static final VERBOSE_LOGGING_SAMPLE = 'ismaster'
 
-    @Rule
-    TemporaryFolder tmp
+	@TempDir
+	File tmp;
```
Comment (roberttaylor426) [inline src/test/groovy/com/sourcemuse/gradle/plugin/MongoPluginConfigSpec.groovy:23]: This project tends not to use semi-colons. Could we drop this?

File: src/test/groovy/com/sourcemuse/gradle/plugin/MongoPluginTasksSpec.groovy [hunk 6]
```diff
@@ -1,22 +1,22 @@
...
 
 class MongoPluginTasksSpec extends Specification {
 
     static final String MONGO_STARTED_MESSAGE = 'Mongod started'
     static final String STOPPING_MONGO_MESSAGE = 'Stopping Mongod'
 
-    @Rule
-    TemporaryFolder tmp
+	@TempDir
+	File tmp;
```
Comment (roberttaylor426) [inline src/test/groovy/com/sourcemuse/gradle/plugin/MongoPluginTasksSpec.groovy:19]: Semi-colon

File: src/main/groovy/com/sourcemuse/gradle/plugin/flapdoodle/gradle/GradleMongoPlugin.groovy [hunk 7]
```diff
@@ -76,11 +77,28 @@ class GradleMongoPlugin implements Plugin<Project> {
...
+				@Override
+				public void projectsEvaluated(Gradle gradle) {}
+			
+				@Override
+				public void projectsLoaded(Gradle gradle) {}
+			
+				@Override
+				public void settingsEvaluated(Settings gradle) {}
+								
+			})
```
Comment (roberttaylor426) [inline src/main/groovy/com/sourcemuse/gradle/plugin/flapdoodle/gradle/GradleMongoPlugin.groovy:101]: Indentation

File: src/main/groovy/com/sourcemuse/gradle/plugin/flapdoodle/adapters/ProcessOutputFactory.groovy [hunk 8]
```diff
@@ -33,15 +35,20 @@ class ProcessOutputFactory {
...
 
-            return new ProcessOutput(
-                    new NamedOutputStreamProcessor('[mongod output]', fileOutputStreamProcessor),
-                    new NamedOutputStreamProcessor('[mongod error]', fileOutputStreamProcessor),
-                    new NamedOutputStreamProcessor('[mongod commands]', fileOutputStreamProcessor));
+            return ProcessOutput.builder()
+				.output(new NamedOutputStreamProcessor('[mongod output]', fileOutputStreamProcessor))
+                .error(new NamedOutputStreamProcessor('[mongod error]', fileOutputStreamProcessor))
+                .commands(new NamedOutputStreamProcessor('[mongod commands]', fileOutputStreamProcessor))
+				.build();
```
Comment (roberttaylor426) [inline src/main/groovy/com/sourcemuse/gradle/plugin/flapdoodle/adapters/ProcessOutputFactory.groovy:42]: Indentation

**Comment by roberttaylor426** [comment]
Hi! :wave: 

Thanks for the PR, it looks great.

Pretty much all of my comments are superficial, almost awkwardly so :joy:. I'm on the road at the moment; if you could address them then when I'm back in the office I should be all ready to cut a release.

File: build.gradle (hunk 1 above)
Comment (flashboss) [reply build.gradle:83]: have you not an email?

File: src/main/groovy/com/sourcemuse/gradle/plugin/flapdoodle/gradle/GradleMongoPlugin.groovy (hunk 7 above)
Comment (flashboss) [reply src/main/groovy/com/sourcemuse/gradle/plugin/flapdoodle/gradle/GradleMongoPlugin.groovy:101]: in my Eclipse environment all is ok. I don't know how to correct it

File: src/main/groovy/com/sourcemuse/gradle/plugin/flapdoodle/adapters/CustomFlapdoodleRuntimeConfig.groovy (hunk 3 above)
Comment (flashboss) [reply src/main/groovy/com/sourcemuse/gradle/plugin/flapdoodle/adapters/CustomFlapdoodleRuntimeConfig.groovy:71]: in my Eclipse environment all is ok. I don't know how to correct it

File: build.gradle (hunk 2 above)
Comment (flashboss) [reply build.gradle:66]: i would like to try it locally but i don't know very well gradle to be able to do a maven publication locally. Actually only tests are tested

**Comment by flashboss** [comment]
Triing to call the plugin by an external client but without success.

When I deploy locally the plugin through the command: ./gradlew publishToMavenLocal I get in my local repository 

 find .m2/repository/com/sourcemuse                                                                                         
.m2/repository/com/sourcemuse
.m2/repository/com/sourcemuse/gradle
.m2/repository/com/sourcemuse/gradle/plugin
.m2/repository/com/sourcemuse/gradle/plugin/gradle-mongo-plugin
.m2/repository/com/sourcemuse/gradle/plugin/gradle-mongo-plugin/1.0.8-SNAPSHOT
.m2/repository/com/sourcemuse/gradle/plugin/gradle-mongo-plugin/1.0.8-SNAPSHOT/maven-metadata-local.xml
.m2/repository/com/sourcemuse/gradle/plugin/gradle-mongo-plugin/1.0.8-SNAPSHOT/gradle-mongo-plugin-1.0.8-SNAPSHOT.pom
.m2/repository/com/sourcemuse/gradle/plugin/gradle-mongo-plugin/maven-metadata-local.xml
.m2/repository/com/sourcemuse/gradle/plugin/GradleMongoPlugin
.m2/repository/com/sourcemuse/gradle/plugin/GradleMongoPlugin/1.0.8-SNAPSHOT
.m2/repository/com/sourcemuse/gradle/plugin/GradleMongoPlugin/1.0.8-SNAPSHOT/GradleMongoPlugin-1.0.8-SNAPSHOT.module
.m2/repository/com/sourcemuse/gradle/plugin/GradleMongoPlugin/1.0.8-SNAPSHOT/GradleMongoPlugin-1.0.8-SNAPSHOT.jar
.m2/repository/com/sourcemuse/gradle/plugin/GradleMongoPlugin/1.0.8-SNAPSHOT/maven-metadata-local.xml
.m2/repository/com/sourcemuse/gradle/plugin/GradleMongoPlugin/1.0.8-SNAPSHOT/GradleMongoPlugin-1.0.8-SNAPSHOT.pom
.m2/repository/com/sourcemuse/gradle/plugin/GradleMongoPlugin/maven-metadata-local.xml
.m2/repository/com/sourcemuse/mongo
.m2/repository/com/sourcemuse/mongo/com.sourcemuse.mongo.gradle.plugin
.m2/repository/com/sourcemuse/mongo/com.sourcemuse.mongo.gradle.plugin/1.0.8-SNAPSHOT
.m2/repository/com/sourcemuse/mongo/com.sourcemuse.mongo.gradle.plugin/1.0.8-SNAPSHOT/com.sourcemuse.mongo.gradle.plugin-1.0.8-SNAPSHOT.pom
.m2/repository/com/sourcemuse/mongo/com.sourcemuse.mongo.gradle.plugin/1.0.8-SNAPSHOT/maven-metadata-local.xml
.m2/repository/com/sourcemuse/mongo/com.sourcemuse.mongo.gradle.plugin/maven-metadata-local.xml

Here the build.gradle file of the client:

...
    repositories {
		mavenLocal()
    }
...
plugins {
  id 'com.sourcemuse.mongo' version '1.0.8-SNAPSHOT'
}
...

I receive the error:

 ./gradlew stopMongoDb

FAILURE: Build failed with an exception.

* Where:
Build file '...../build.gradle' line: 15

* What went wrong:
Plugin [id: 'com.sourcemuse.mongo', version: '1.0.8-SNAPSHOT'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Plugin Repositories (could not resolve plugin artifact 'com.sourcemuse.mongo:com.sourcemuse.mongo.gradle.plugin:1.0.8-SNAPSHOT')
  Searched in the following repositories:
    Gradle Central Plugin Repository

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Have you some idea?

**Comment by roberttaylor426** [comment]
Does this [SO answer](https://stackoverflow.com/questions/51773218/why-does-gradle-not-look-in-the-local-maven-repository-for-plugins) help?

**Comment by flashboss** [comment]
junit testsa re ok but how do you test locally the final release? Have you find problems?

**Comment by flashboss** [comment]
the answer helped me. Thank you. I tested the plugin through the external client and mongo starts with the last version. So I think this branch can be released.

**Comment by flashboss** [comment]
Are there news?

**Comment by roberttaylor426** [comment]
Pushed v2.0.0 today. You should see it appear on the Gradle Plugins repo and Maven Central shortly.

**Comment by flashboss** [comment]
I find only 1.0.7 version here https://plugins.gradle.org/search?term=mongo

**Comment by roberttaylor426** [comment]
When I published 2.0.0 to the Gradle plugins repo, the Gradle team advised that we might need to wait a day or two for it to appear.

**Comment by flashboss** [comment]
also mvnrepository is not updated https://mvnrepository.com/artifact/com.sourcemuse.gradle.plugin/gradle-mongo-plugin

**Comment by flashboss** [comment]
in https://repo.maven.apache.org/maven2/com/sourcemuse/gradle/plugin/gradle-mongo-plugin/ is ok!

**Comment by flashboss** [comment]
Still nothing! https://mvnrepository.com/artifact/com.sourcemuse.mongo/com.sourcemuse.mongo.gradle.plugin https://plugins.gradle.org/plugin/com.sourcemuse.mongo

**Comment by roberttaylor426** [comment]
I'm not sure why we're not seeing anything in `mvnrepository.com` yet. I'm speaking to someone at Gradle about publication to the plugins repo, they came back with a couple of questions. Will keep you posted.

**Comment by flashboss** [comment]
Now all is ok. I downloaded the gradle plugin and started mongodb 5..0.6 !!!

**Comment by roberttaylor426** [comment]
Excellent!

**Pull request merged**